Taichow keyboard updating

Cathaylaboratory opened the pull request:
ready for reviewing

keyman-server commented:
Thank you for your pull request. You'll see a "build failed" message until the Keyman team has reviewed the pull request and manually initiated the build process.

Every change committed to this branch will become part of this pull request. When you have finished submitting files and are ready for the Keyman team to review this pull request, please post a "Ready for review" comment.

LornaSIL commented:
You've reverted some of the changes we made previously.
In particular:
- `LICENSE.md` needs the copyright span of `2024-2025`, not just `2024`. Please adjust that.
- In `taichow.php`, `readme.htm` and `welcome.htm` we had removed the copyright year so we wouldn't have to keep updating that field. Please remove the year. The only place we want the year included is in `LICENSE.md`.

Please make sure you always update the files from the repo before you begin to make changes. That way this kind of thing won't happen.

LornaSIL commented:
Thanks. Remember the .user file should never be committed as that's a transient file.

Cathaylaboratory commented:
I deleted it

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Saturday, 17 May 2025 at 5:16 PM, Lorna Evans ***@***.***> wrote:

> LornaSIL left a comment [(keymanapp/keyboards#3522)](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2888477427)
>
> Thanks. Remember the .user file should never be committed as that's a transient file.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2888477427), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BCAJHRWDEH54VWX4GFQT7VT265OHFAVCNFSM6AAAAAB5FCWC6GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOBYGQ3TONBSG4).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

Cathaylaboratory commented:
Can you approve the request? I have deleted the user file

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Saturday, 17 May 2025 at 5:16 PM, Lorna Evans ***@***.***> wrote:

> LornaSIL left a comment [(keymanapp/keyboards#3522)](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2888477427)
>
> Thanks. Remember the .user file should never be committed as that's a transient file.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2888477427), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BCAJHRWDEH54VWX4GFQT7VT265OHFAVCNFSM6AAAAAB5FCWC6GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOBYGQ3TONBSG4).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

LornaSIL commented:
I am not ready to approve it. I'm asking someone else to review the `taichow_ime.js` file before this is approved.

Cathaylaboratory commented:
I think I need to delete this part because I failed to make it.
The current codes only support korean jamo, not for Romanization.
Give me a little bit time to delete it.

Sent from Proton Mail for iOS

On Mon, May 19, 2025 at 22:37, Lorna Evans ***@***.***> wrote:

> LornaSIL left a comment [(keymanapp/keyboards#3522)](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2892327584)
>
> I am not ready to approve it. I'm asking someone else to review the taichow_ime.js file before this is approved.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/keymanapp/keyboards/pull/3522#issuecomment-2892327584), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BCAJHRUU3RFNAXXOUALWKGT27JFKHAVCNFSM6AAAAAB5FCWC6GVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQOJSGMZDONJYGQ).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

jahorton commented on release/t/taichow/source/taichow_ime.js:
```diff
@@ -0,0 +1,983 @@
+  k.hide();
+}
+
+this.hanja = {
```
The data here appears to be the same as for the `korean_rr` keyboard, but this is the spot that _really_ should be tailored with data for your language if trying to make an IME picker for it.

Or does it actually use / map to the same script?

jahorton replied:
When I copy this entire file and replace `korean_rr`'s original implementation with it, there _are_ some entries below that show "changes" - but so far as I can tell, the characters still match the original visibly.

jahorton commented on release/t/taichow/source/taichow_ime.js:
I see that this is based upon the `korean_rr` keyboard's IME - the code is near identical outside of replacing a few keyboard-id references to match its new home.

Note that this style of IME only works on desktop form-factor devices; the IME will neither activate nor show up on phones or tablets (unless emulating a desktop device).

Cathaylaboratory commented:
I deleted it.
The korean rr keyboard is one jamo to one kanji transform. Mine is romanization transformation, so mine will be different with korean rr keyboard.
I have deleted the js file. Please approve the request if everything is OK

Sent with [Proton Mail](https://proton.me/mail/home) secure email.

On Tuesday, 20 May 2025 at 2:29 AM, Joshua Horton ***@***.***> wrote:

> @jahorton commented on this pull request.
>
> ---------------------------------------------------------------
>
> In [release/t/taichow/source/taichow_ime.js](https://github.com/keymanapp/keyboards/pull/3522#discussion_r2096682380):
>
>> +    a = KeymanWeb.GetLastActiveElement();
> +    KeymanWeb.KT(null);    /* KT will refocus the active element in the case of a mouse click */
> +    c = KeymanWeb.KC(12,12,a);
> +    cLen = c.length;
> +
> +    /* Find the part of the context matching hangul for the currently displayed list */
> +    for(j=1; j<=cLen; j++)
> +    {
> +      if(k.getHangul(c.substr(cLen-j)) == tip) break;
> +    }
> +    if( j <= cLen) KeymanWeb.KO(j,a,chr);
> +  }
> +  k.hide();
> +}
> +
> +this.hanja = {
>
> The data here appears to be the same as for the korean_rr keyboard, but this is the spot that really should be tailored with data for your language if trying to make an IME picker for it.
>
> Or does it actually use / map to the same script?
>
> ---------------------------------------------------------------
>
> On [release/t/taichow/source/taichow_ime.js](https://github.com/keymanapp/keyboards/pull/3522#discussion_r2096687296):
>
> I see that this is based upon the korean_rr keyboard's IME - the code is near identical outside of replacing a few keyboard-id references to match its new home.
>
> Note that this style of IME only works on desktop form-factor devices; the IME will neither activate nor show up on phones or tablets (unless emulating a desktop device).
>
> ---------------------------------------------------------------
>
> In [release/t/taichow/source/taichow_ime.js](https://github.com/keymanapp/keyboards/pull/3522#discussion_r2096688214):
>
>> +    a = KeymanWeb.GetLastActiveElement();
> +    KeymanWeb.KT(null);    /* KT will refocus the active element in the case of a mouse click */
> +    c = KeymanWeb.KC(12,12,a);
> +    cLen = c.length;
> +
> +    /* Find the part of the context matching hangul for the currently displayed list */
> +    for(j=1; j<=cLen; j++)
> +    {
> +      if(k.getHangul(c.substr(cLen-j)) == tip) break;
> +    }
> +    if( j <= cLen) KeymanWeb.KO(j,a,chr);
> +  }
> +  k.hide();
> +}
> +
> +this.hanja = {
>
> When I copy this entire file and replace korean_rr's original implementation with it, there are some entries below that show "changes" - but so far as I can tell, the characters still match the original visibly.
>
> —
> Reply to this email directly, [view it on GitHub](https://github.com/keymanapp/keyboards/pull/3522#pullrequestreview-2852283850), or [unsubscribe](https://github.com/notifications/unsubscribe-auth/BCAJHRVSALI2ALL53D2IHV327KAQ5AVCNFSM6AAAAAB5FCWC6GVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDQNJSGI4DGOBVGA).
> You are receiving this because you authored the thread.Message ID: ***@***.***>

DavidLRowe commented:
Thanks for the update. 

Keyboard review is currently on hold until some issues with the automated build system are resolved.

We appreciate your patience.

LornaSIL commented:
The keyboards in the `release` section of our keyboard repo are considered stable and should not be constantly changing. I think adding a few rules to a keyboard is consistent with being in `release`, but constantly adding more and more rules is not consistent with how `release` should be used. I'm going to close this PR. 

Any of your future development of keyboards should go into `experimental` while you continue to test what is a good layout for your users. However, for this kind of thing trying to implement an IME is not appropriate. We will just have to wait until the Keyman team has the resources to implement a proper IME which is going to be some time in the future.
